src/include/sof/drivers/sdma.h

```diff
@@ -164,7 +164,183 @@
 #define SDMA_SCRIPT_SAI2MCU_OFF		6710
 
 #if CONFIG_HAVE_SDMA_FIRMWARE
-#include "sdma_script_code_imx7d_4_5.h"
+
+/*
+ * From "sdma_script_code_imx7d_4_5.h"
+ */
+
+/* SDMA RAM image start address and size */
+#define RAM_CODE_START_ADDR     6144
+#define RAM_CODE_SIZE           1315
+
+/* buffer that holds the SDMA RAM image */
+static const short sdma_code[] = {
```

Comment:
Suggestion: maybe we should encase this in an `#ifdef` block? I'm assuming we're only going to use this if we need to change the "default" script. For MICFIL we could do something like `select SDMA_ALLOW_SCRIPT_CHANGE` in `config IMX_MICFIL`.

Reply:
For now this can be opted out using CONFIG_HAVE_SDMA_FIRMWARE. So if you want to change the default script just select CONFIG_HAVE_SDMA_FIRMWARE.

Reply:
My bad, missed the already existing `#if CONFIG_HAVE_SDMA_FIRMWARE` block.